pkg/app/api/grpcapi/piped_api.go

```diff
@@ -957,8 +957,24 @@ func (a *PipedAPI) GetDesiredVersion(ctx context.Context, _ *pipedservice.GetDes
 }
 
 func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pipedservice.UpdateApplicationConfigurationsRequest) (*pipedservice.UpdateApplicationConfigurationsResponse, error) {
-	// TODO: Update the given application configurations
-	return nil, status.Errorf(codes.Unimplemented, "UpdateApplicationConfigurations is not implemented yet")
+	_, _, _, err := rpcauth.ExtractPipedToken(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
since we have `appInfo.Id`, better to validate if the updating application belongs to the requested piped as other RPCs.

Reply:
For sure

Reply:
Applied at d22b018